src/dotnet/OpenAPI.Validator/Model/Utilities/ValidationUtilities.cs

```diff
@@ -365,33 +365,14 @@ private static IEnumerable<Operation> SelectOperationsFromPaths(string id, Dicti
         /// <summary>
         /// Returns a suggestion of camel case styled string based on the string passed as parameter.
         /// </summary>
+        /// <example>
+        /// <code>URI</code> becomes <code>Uri</code>
+        /// <code>SomeURIThing</code> becomes <code>SomeUriThing</code>
+        /// </example>
         /// <param name="name">String to convert to camel case style</param>
         /// <returns>A string that conforms with camel case style based on the string passed as parameter.</returns>
         public static string GetCamelCasedSuggestion(string name)
-        {
-            StringBuilder sb = new StringBuilder(name);
-            if (sb.Length > 0)
-            {
-                sb[0] = sb[0].ToString().ToLower()[0];
-            }
-            bool firstUpper = true;
-            for (int i = 1; i < name.Length; i++)
-            {
-                if (char.IsUpper(sb[i]) && firstUpper)
-                {
-                    firstUpper = false;
-                }
-                else
-                {
-                    firstUpper = true;
-                    if (char.IsUpper(sb[i]))
-                    {
-                        sb[i] = sb[i].ToString().ToLower()[0];
-                    }
-                }
-            }
-            return sb.ToString();
-        }
+            => Regex.Replace(name, @"(?<=\p{Lu})\p{Lu}+(?=\p{Lu}|\z)", s => s.Value.ToLowerInvariant());
```

Comment:
I'm all for regexes but even I can't comprehend this on the first go. I'd keep the dirty logic as is since this is bound to evolve further. @veronicagg thoughts?

Reply:
note that the logic is *changed* to actually alter the rule, so there's not much to "keep". But I agree, there may be value in rewriting this Regex as code (preferably not dirty, but readable 😄 ) for future maintainability

Reply:
Ahh, did not notice that, but yes, still see value in turning the regex into code.

Reply:
Yeah, the reason I haven't reviewed this PR is because I had to set time aside to study the regex :) if we can put it in simpler terms it'd be great, if not, may be an explanation of the regex in comments would help. The current logic suggesting a string is not ideal, but that's the reason we kept it as a suggestion, it was a way to "show" the issue and a compliant option more than having people take the suggestion.

Reply:
It's _quite_ clunky if converted to code but I'll comment the regex to explain what it's doing

Reply:
... done that, and added a test for expected behaviour.